src/main/java/net/snowflake/ingest/streaming/internal/BlobBuilder.java

```diff
@@ -61,10 +61,14 @@ class BlobBuilder {
    * @param blobData All the data for one blob. Assumes that all ChannelData in the inner List
    *     belongs to the same table. Will error if this is not the case
    * @param bdecVersion version of blob
+   * @param disableChunkEncryption If the encryption is disabled
    * @return {@link Blob} data
    */
   static <T> Blob constructBlobAndMetadata(
-      String filePath, List<List<ChannelData<T>>> blobData, Constants.BdecVersion bdecVersion)
+      String filePath,
+      List<List<ChannelData<T>>> blobData,
+      Constants.BdecVersion bdecVersion,
+      boolean disableChunkEncryption)
```

Comment:
nit: suggest calling it `boolean encrypt` and pass in true for non-iceberg / false for iceberg.